File: leetcode/python/kidsWithTheGreatest.py

def kidsWithCandies(candies, extraCandies):
    max_value = max(candies)
    print(max_value)
    result = []
    for i in candies:
        if i + extraCandies >= max_value:
            result.append('true')
        else:
            result.append('false')
    return result


print(kidsWithCandies([2, 3, 5, 1, 3], 3))

File: leetcode/javascript/kidsWithTheGreatest.js
var kidsWithCandies = function (candies, extraCandies) {
  const result = [];
  let max = 0;
  for (let i = 0; i < candies.length; i++) {
    if (candies[i] >= max) {
      max = candies[i];
    }
  }

  for (let i = 0; i < candies.length; i++) {
    if (candies[i] + extraCandies >= max) {
      result.push(true);
    } else {
      result.push(false);
    }
  }

  return result;
};
